src/server/http_routes.rs

```diff
@@ -219,27 +282,41 @@ async fn parse_body(req: hyper::Request<Body>) -> Result<Bytes, ApplicationError
     }
 }
 
+macro_rules! return_err {
+    ($val:expr) => {{
+        match $val {
+            Ok(v) => v,
+            Err(e) => return e.response(),
+        }
+    }};
+}
```

Comment:
```suggestion
macro_rules! return_err {
    ($val:expr) => {{
        match $val {
            Ok(v) => v,
            Err(e) => {
                error!(error = ?e, error_message = ?e.to_string(), "Error while handling request");
                return e.response()
            },
        }
    }};
}
```

When I was trying this out locally, I noticed that errors didn't make it into the log (as in there was no `ERROR` log entry). I think that is pretty critical so recommend adding this

Reply:
The `error_handler` function, as explained above, doesn't seem to be invoked in this case